src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -263,169 +269,67 @@ static uint8_t readLevelControlCurrentLevel(EndpointId endpoint)
-                                                              uint16_t transitionTime, uint8_t optionsMask, uint8_t optionsOverride)
-{
-    EndpointId endpoint = emberAfCurrentEndpoint();
-    uint8_t currentHue  = readHue(endpoint);
-    bool moveUp;
-
-    if (transitionTime == 0)
-    {
-        transitionTime++;
-    }
-
-    // limit checking:  hue and saturation are 0..254.  Spec dictates we ignore
-    // this and report a malformed packet.
-    if (hue > MAX_HUE_VALUE || saturation > MAX_SATURATION_VALUE)
-    {
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_MALFORMED_COMMAND);
-        return true;
-    }
-
-    if (!shouldExecuteIfOff(endpoint, optionsMask, optionsOverride))
-    {
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-        return true;
-    }
-
-    // compute shortest direction
-    if (hue > currentHue)
-    {
-        moveUp = (hue - currentHue) < HALF_MAX_UINT8T;
-    }
-    else
-    {
-        moveUp = (currentHue - hue) > HALF_MAX_UINT8T;
-    }
-
-    // New command.  Need to stop any active transitions.
-    stopAllColorTransitions();
-
-    // Handle color mode transition, if necessary.
-    handleModeSwitch(endpoint, COLOR_MODE_HSV);
-
-    // now, kick off the state machine.
-    initHueSat(endpoint);
-
-    colorHueTransitionState.initialHue     = currentHue;
-    colorHueTransitionState.currentHue     = currentHue;
-    colorHueTransitionState.finalHue       = hue;
-    colorHueTransitionState.stepsRemaining = transitionTime;
-    colorHueTransitionState.stepsTotal     = transitionTime;
-    colorHueTransitionState.endpoint       = endpoint;
-    colorHueTransitionState.up             = moveUp;
-    colorHueTransitionState.repeat         = false;
-
-    colorSaturationTransitionState.initialValue   = readSaturation(endpoint);
-    colorSaturationTransitionState.currentValue   = readSaturation(endpoint);
-    colorSaturationTransitionState.finalValue     = saturation;
-    colorSaturationTransitionState.stepsRemaining = transitionTime;
-    colorSaturationTransitionState.stepsTotal     = transitionTime;
-    colorSaturationTransitionState.endpoint       = endpoint;
-    colorSaturationTransitionState.lowLimit       = MIN_SATURATION_VALUE;
-    colorSaturationTransitionState.highLimit      = MAX_SATURATION_VALUE;
 
-    writeRemainingTime(endpoint, transitionTime);
-
-    // kick off the state machine:
-    emberEventControlSetDelayMS(&COLOR_HSV_CONTROL, UPDATE_TIME_MS);
-
-    emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-    return true;
-}
-
-bool emberAfColorControlClusterMoveHueCallback(chip::app::Command * commandObj, uint8_t moveMode, uint8_t rate, uint8_t optionsMask,
-                                               uint8_t optionsOverride)
+static bool moveHue(uint8_t moveMode, uint16_t rate, uint8_t optionsMask, uint8_t optionsOverride, bool isEnhanced)
```

Comment:
Perhaps document that when `isEnhanced` is true `rate` will be in the `uint8_t` range?